project-metrics/metrics_service/database/cherrypick_scraper.py

```diff
@@ -0,0 +1,75 @@
+from typing import Sequence
+import logging
+import time
+import sqlalchemy
+
+from apis import github
+from database import db
+from database import models
+
+SCRAPE_INTERVAL_SECONDS = 3
+
+
+class CherrypickScraper(object):
+
+  def __init__(self):
+    self.github = github.GitHubGraphQL()
+    self.session = db.Session()
+
+    commits = self.session.query(models.Commit.hash).union(
+        self.session.query(models.Cherrypick.hash)).all()
+    self.seen_commit_hashes = set(commit.hash for commit in commits)
+
+  def scrape_release_cherrypicks(
+      self, release: models.Release) -> Sequence[models.Cherrypick]:
+    """Determines the commit SHAs of cherrypick commits in a release.
+
+    This is determined by identifying all the commits in the release tag ref
+    beyond the merge-base.
+
+    Args:
+      release: release to scrape cherry-picks for.
+
+    Returns:
+      A list of cherrypicks
+    """
+    logging.info('Querying GitHub for commits in release "%s"', release.name)
+    response = self.github.query_repo("""
+      release(tagName: "{release_name}") {{
+        tag {{
+          target {{
+            ... on Commit {{
+              history(first: 20) {{
```

Comment:
Magic number?

Reply:
Just a (very) loose upper bound on how many commits to consider when finding cherrypicks. Since this is more than the total number of cherrypicks in the last 90 days, it seems unlikely there will every be 20 cherrypicks in one release

Reply:
Fair, but might still be worth documenting it in code and maybe making it a const too